openmls/src/group/core_group/new_from_welcome.rs

```diff
@@ -58,16 +58,15 @@ impl CoreGroup {
             return Err(e);
         }
 
-        let group_secrets_bytes = backend
-            .crypto()
-            .hpke_open(
-                ciphersuite.hpke_config(),
-                egs.encrypted_group_secrets(),
-                key_package_bundle.private_key.as_slice(),
-                &[],
-                &[],
-            )
-            .map_err(|_| WelcomeError::UnableToDecrypt)?;
+        let group_secrets_bytes = hpke::decrypt_with_label(
+            key_package_bundle.private_key.as_slice(),
+            "Welcome",
```

Comment:
Maybe not in this PR, but it would probably make sense to have a dedicated type for `InitKey` in the same way we already have for `Encryptionkey`.